### DISCUSSION TITLE
emsdk_env.bat is missing

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#9467] [Q&A]
[_From Sentei#5306 on Discord_: ](https://discord.com/channels/372137812037730304/550416151172087808/1007079990434742313)

I reinstalled windows, trying to get my wasm project running again from 0, followed the getting started guide, running into emsdk_env.bat file not being where it's supposed to be, any thought's on what I could do to fix?

It looks like this is related to using skiasharp, creating a new project and adding 
`<PackageReference Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.1-preview.63" />`
 causes me to be unable to build wasm with any kind of settings, so I'm a bit stuck now.

**Comment by jeromelaban** [comment]
This generally means that emscripten failed to install in a previous run, mostly likely because [python is not installed](https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/runtime-execution-modes.html?q=python#runtime-execution-modes).

Once you've installed python, you'll likely need to delete the `%USERPROFILE%\.uno\emsdk` folder and try to build again.

**Reply by Rytisgit** [reply]
Thanks for the suggestion, I did receive an explicit python is not installed error initially before asking for help and installed it, this is already with python 3.7.9 installed, unless a newer version is needed I don't think it's the problem. I have also tried to delete the folder as suggested, but there was no difference.

I tried to create a new app from the template and adding my current project nuggets, it works if skiasharp is removed, doesn't build if ```<PackageReference Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.1-preview.104" />``` is included, builds if ```<!--<PackageReference Include="SkiaSharp.Skottie" Version="2.88.1-preview.104" />-->``` is included, but calling skiasharp when the app is launched results in `System.DllNotFoundException: libSkiaSharp at SkiaSharp.SKImageInfo..cctor() in D:\a\1\s\binding\Binding\SKImageInfo.cs:line 46` same errors as I reported in https://github.com/unoplatform/Uno.Wasm.Bootstrap/issues/542 and https://github.com/unoplatform/Uno.Wasm.Bootstrap/issues/543 but now it seems even switching ILLinker enabled to true doesn't allow it to build or not settings threads for it to work.
[VS2022SkiaError.zip](https://github.com/unoplatform/uno/files/9309138/VS2022SkiaError.zip)

**Reply by jeromelaban** [reply]
Thanks. Can you provide a binlog of the build: https://platform.uno/docs/articles/uno-builds-troubleshooting.html?

**Reply by Rytisgit** [reply]
Sure, this is the failing build with ```<PackageReference Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.1-preview.104" />```
[msbuild.zip](https://github.com/unoplatform/uno/files/9310050/msbuild.zip)

**Reply by jeromelaban** [reply]
Thanks. Two things:
- Could you delete the `%USERPROFILE%\.uno\emsdk` then immediately rerun a build and create another binlog file with it?
- Does the `%USERPROFILE%\.uno\emsdk` contain that file? I know that by default, git is not installed by visual studio on the command line, and I wonder if that could be the issue.

**Reply by Rytisgit** [reply]
Thanks a bunch, your hunch about git was correct, after deleting the folder it did display an error about not finding git in the binlog(which I included). I installed git, deleted the folder again to get it to rebuild, then it showed another error about powershell not having the execution-policy set to run the script also in a binlog, which I had to set to bypass. After these 2 changes I am back in action. I am feeling a bit stupid about not having git installed, but in my defense it is not mentioned as necessary in the getting started guide. Thanks again for the help.
[msbuild.zip](https://github.com/unoplatform/uno/files/9311616/msbuild.zip)

**Reply by jeromelaban** [reply]
That's correct, it's not mentioned anywhere and we'll correct that, it'll be included as a prerequisite of uno-check. Thanks for the update!
